src/Workleap.DomainEventPropagation.Subscription.PullDelivery/CloudEventHandler.cs

```diff
@@ -16,32 +16,66 @@ internal sealed class CloudEventHandler : BaseEventHandler, ICloudEventHandler
         : base(serviceProvider, domainEventTypeRegistry)
     {
         this._logger = logger;
-        this._pipeline = domainEventBehaviors.Reverse().Aggregate((DomainEventHandlerDelegate)HandleDomainEventAsync, BuildPipeline);
+        this._pipeline = domainEventBehaviors.Reverse().Aggregate((DomainEventHandlerDelegate)this.HandleDomainEventAsync, BuildPipeline);
     }
 
     public async Task<EventProcessingStatus> HandleCloudEventAsync(CloudEvent cloudEvent, CancellationToken cancellationToken)
     {
-        var domainEventWrapper = new DomainEventWrapper(cloudEvent);
+        try
+        {
+            if (!TryWrapCloudEvent(cloudEvent, out var domainEventWrapper))
+            {
+                this._logger.IllFormedCloudEvent(cloudEvent.Id);
+                return EventProcessingStatus.Rejected;
+            }
+
+            if (this.GetDomainEventType(domainEventWrapper!.DomainEventName) == null)
+            {
+                this._logger.EventDomainTypeNotRegistered(domainEventWrapper.DomainEventName, cloudEvent.Subject ?? "Unknown");
+                return EventProcessingStatus.Rejected;
+            }
 
-        if (this.GetDomainEventType(domainEventWrapper!.DomainEventName) == null)
+            return await this._pipeline(domainEventWrapper, cancellationToken).ConfigureAwait(false);
+        }
+        catch (Exception e)
         {
-            this._logger.EventDomainTypeNotRegistered(domainEventWrapper.DomainEventName, cloudEvent.Subject ?? "Unknown");
-            return EventProcessingStatus.Rejected;
+            this._logger.EventHandlingFailed(cloudEvent.Type, cloudEvent.Id, e.Message);
+            return EventProcessingStatus.Released;
         }
+    }
 
-        return await this._pipeline(domainEventWrapper, cancellationToken).ConfigureAwait(false);
+    private static bool TryWrapCloudEvent(CloudEvent cloudEvent, out DomainEventWrapper? domainEventWrapper)
+    {
+        domainEventWrapper = null;
+        try
+        {
+            domainEventWrapper = new DomainEventWrapper(cloudEvent);
+            return true;
+        }
+        catch (Exception e)
```

Comment:
Should we log the exception as in error here?

If we don't, you should not need to declare any variable for it:
```suggestion
        catch (Exception)
```